include/partition/inertial_flow.hpp

```diff
@@ -2,20 +2,56 @@
 #define OSRM_PARTITION_INERTIAL_FLOW_HPP_
 
 #include "partition/graph_view.hpp"
+
+#include <unordered_set>
 #include <vector>
 
 namespace osrm
 {
 namespace partition
 {
 
+// TODO: dummy
+
+struct MinCut
+{
+    std::int32_t num_edges;
+    // Should actually contain cut edges and/or lhs and rhs nodes
+};
+
+inline MinCut Dinic(const GraphView &graph,
+                    const std::unordered_set<NodeID> &sources,
+                    const std::unordered_set<NodeID> &sinks)
+{
+    return MinCut{10};
+}
+
+//
+
 class InertialFlow
 {
   public:
     InertialFlow(const GraphView &view);
 
     std::vector<bool> ComputePartition(const double balance, const double source_sink_rate);
+
   private:
+    // Spatially ordered sources and sink ids.
+    // The node ids refer to nodes in the GraphView.
+    struct SpatialOrder
+    {
+        std::unordered_set<NodeID> sources;
+        std::unordered_set<NodeID> sinks;
+    };
+
+    // Creates a spatial order of n * sources "first" and n * sink "last" node ids.
+    // The slope determines the spatial order for sorting node coordinates.
+    SpatialOrder makeSpatialOrder(double ratio, double slope) const;
```

Comment:
Capturing some trade-offs and ideas from chat here:
- Our typical `ratio` will be around .25 which means we have to hold .25\*nodes + .25\*nodes for sources and sinks, respectively, in memory. This can get quite large for planet sized graphs.
- We could re-order the nodes in-place (in the GraphView) but at the moment it is immutable and we want to keep it that way, so that we can run partitioning in parallel. We could `lock(); reorder(); unlock()` in the GraphView directly but this makes our abstractions ugly.